### PR TITLE
feat: show tooltips indicating the source of config values

### LINKED ISSUE #21830: App - Settings Page - Config tooltips
## **Summary**
In 9.x, in addition to the legend, we show tooltips indicating the source of values in the in the resolved config, when you hover over the values. We should match this behavior in 10.0

These are useful:

* when the legend is out of view due to scrolling the page
* for color blind users, where the color-only nature of the legend may not be functional
* weirdos like me who sometimes put the whole operating system in grayscale

## **Acceptance Criteria**
* Show tooltip on hover of each config value describing where it was set.
* Write a test ensuring the tooltip is shown on hover.

### **Resources**
9.x behavior

Screen Shot 2022-04-20 at 11 16 48 AM

PjDNrX1uZt

┆Issue is synchronized with this [Jira Task](https://cypress-io.atlassian.net/browse/UNIFY-1573) by [Unito](https://www.unito.io)
┆Attachments: <a href="https://cypress-io.atlassian.net/rest/api/2/attachment/content/11805">Screen Shot 2022-04-20 at 11.16.48 AM.png</a>
┆author: Mark Noonan
┆epic: 9.x Parity
┆friendlyId: UNIFY-1573
┆priority: Medium
┆sprint: Backlog
┆taskType: Task

### PR DESCRIPTION
**Pull request by gitstart**
- Closes https://github.com/cypress-io/cypress/issues/21830

### User facing changelog

Fix an issue were user does not see a tooltip showing the source of the config value on hover. Fixes #21830

### Additional details
In 9.x, in addition to the legend, we show tooltips indicating the source of values in the in the resolved config, when you hover over the values. We should match this behavior in 10.0.

### Steps to test
- Prepare/view target project
- Go to project settings page
- In the `Resolved Configuration` section, see a list of configs for the project, listed in a { key: value } pair approah.
- Hover on the value
- Verify that a tooltip is shown which indicates the source of the config value.

### How has the user experience changed?
BEFORE

https://user-images.githubusercontent.com/45232708/184300205-5052b8fe-f7d8-4499-a45c-c5e074b2d976.mov

AFTER

https://user-images.githubusercontent.com/45232708/184299635-325b64c5-7806-4c65-93f4-93ea75a4e6fc.mov

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by gitstart** [comment]
@marktnoonan  could you help to make pipeline run on this branch?

File: packages/app/src/settings/project/ConfigCode.cy.tsx [hunk 1]
```diff
@@ -31,6 +32,34 @@ describe('<ConfigCode />', () => {
...
+          configFileAbsolutePath: '/path/to/cypress.config.js',
+          config: [{
+            field: 'arrayTest',
+            value: arrayTest,
+            from: 'plugin',
+          }],
+        }} />
+      </div>))
+
+      _.each(arrayTest, (val) => {
```
Comment (emilyrohrbough) [inline packages/app/src/settings/project/ConfigCode.cy.tsx:49]:
```suggestion
      Cypress._.each(arrayTest, (val) => {
```

**Comment by gitstart** [comment]
Thanks @emilyrohrbough, requested changes here has been made, kindly take a look.

**Review by ZachJW34 — APPROVED** [review]
Changes look good! I think an improvement here would be to create a reusable component for rendering the primitive values, as the markup looks the same across the same 3 instances the tooltips were implemented.

**Comment by lmiller1990** [comment]
Looks good - WDYT about the above recommendation to refactor into a shared component @gitstart?

**Comment by gitstart** [comment]
Thanks @ZachJW34 @lmiller1990 suggestion sounds good to me, pushing change for it shortly 🙂.

File: packages/app/src/settings/project/renderers/RenderPrimitive.vue [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  class?: string
+}>()
+
+const content = computed(() => {
+  if (props.value === null || props.value === undefined) {
+    return 'null'
+  }
+
+  if (typeof props.value === 'string') {
+    if (props.value.startsWith('[Function')) {
```
Comment (lmiller1990) [inline packages/app/src/settings/project/renderers/RenderPrimitive.vue:34]: Do we have tests for all these different things? Might be good to add a `RendererPrimitive.cy.tsx` spec and use `cy.realHover` to verify the content is actually correct for each branch here.

**Review by ZachJW34 — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]

**Comment by gitstart** [comment]
This PR was pushed through [Gitstart](https://www.gitstart.com/), with contributions from @raph941, @phunguyenmurcul.

**Pull request merged**